week3/homework/src/methodList.js

```diff
@@ -0,0 +1,150 @@
+const fs = require('fs');
+const uuid = require('uuid/v4');
+
+function readFile(path) {
+  return new Promise((resolve, reject) => {
+    fs.readFile(path, 'utf8', (error, data) => {
+      if (error) {
+        reject(error);
+      } else {
+        resolve(data);
+      }
+    });
+  });
+}
+
+function writeFile(path, data) {
+  return new Promise((resolve, reject) => {
+    fs.writeFile(path, JSON.stringify(data), 'utf8', error => {
+      reject(error);
```

Comment:
This will always reject, instead you have to check the error and reject or resolve accordingly; like

    if(error) return reject(error);
    return resolve();